### ISSUE TITLE
NPNP invalid gateway

### ISSUE THREAD

**Issue by BurningFluffer** [godotengine/godot#79857]
### Godot version

4.1.stable

### System information

Godot v4.1.stable - Windows 10.0.19044 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 (NVIDIA; 31.0.15.2824) - Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz (4 Threads)

### Issue description

I'm trying to enable UPNP in my game (create IP adress others would be able to connect to for online multiplayer; setting up the server) and UPNP does get discovered, but the gateway is invalid. I followed a [tutorial](https://youtu.be/n8D3vEx7NAE?t=2451) and they seem to have it working.

### Steps to reproduce

Code so far:
`func upnp_setup():
	var upnp = UPNP.new()
	var discover_result = upnp.discover() 
	assert(discover_result == UPNP.UPNP_RESULT_SUCCESS, "UPNP Discover Failed! Error %s" % discover_result)
	assert(upnp.get_gateway() and upnp.get_gateway().is_valid_gateway(), "UPNP Invalid Gateway!")# <- here is a fail point
	var map_result = upnp.add_port_mapping(PORT)
	assert(map_result == UPNP.UPNP_RESULT_SUCCESS, "UPNP Port Mapping Failed! Error %s" % map_result)
	print("Sucess! Join Address: %s" % upnp.query_external_address())
	$"../../MultiplayerMenu/MPMenu/TextureRect/OwnIP".text = "%s" % upnp.query_external_address()`
Might be router-specific. My router has UPNP enabled. Here are router stats:
![image](https://github.com/godotengine/godot/assets/138900251/5336add0-8557-42d7-8ec0-2df754d49d84)
![image](https://github.com/godotengine/godot/assets/138900251/be407574-9cbe-4057-a26e-9b15b879809b)
![image](https://github.com/godotengine/godot/assets/138900251/d1e5a9ea-5fca-4201-b3dd-0e7e1ef618e4)
![image](https://github.com/godotengine/godot/assets/138900251/676cd552-96f7-4b84-bec7-f334f28968a2)

### Minimal reproduction project

N/A

**Comment by BurningFluffer** [comment]
OK, nevermind. To all who discovers the same issue later: check with your provider. Mine disables UPNP while having it "Enabled" on router, to have "static IP" as a separate payable service. That truly sucks for all (devs with similar providers) and players, because this severs people from creating servers/hosting games. 
Is there any way to work around it using only a dynamic IP? I'm sorry, I'm not an expert with how networking works.

**Comment by Calinou** [comment]
>  Is there any way to work around it using only a dynamic IP? I'm sorry, I'm not an expert with how networking works.

No, other than using dedicated servers and making players create virtual "rooms" on them (or spawning instances on demand). This is also required for players behind CGNAT or players that can't enable UPnP on their router for other reasons (such as not knowing how to do it, or having forgotten login credentials).

This is why UPnP is pretty much dead nowadays in general.
